tangelo/algorithms/variational/vqe_solver.py

```diff
@@ -114,6 +116,7 @@ def __init__(self, opt_dict):
         self.verbose: bool = copt_dict.pop("verbose", False)
         self.projective_circuit: Circuit = copt_dict.pop("projective_circuit", None)
         self.ref_state: Optional[Union[list, Circuit]] = copt_dict.pop("ref_state", None)
+        self.save_energies: bool = copt_dict.pop("save_energies", False)
```

Comment:
I think this design flows well.

Out of curiosity, if we wanted to also monitor other quantities such as S and S^2 for example, how is that achievable ? Is their computation expensive ?

Reply:
This would involve a call to https://github.com/goodchemistryco/Tangelo/blob/293f7610071536ebf0f95076629f74331a79b18e/tangelo/algorithms/variational/vqe_solver.py#L316 at each minimization step. I am expecting this would slow down the process significantly.

Reply:
No, my comment was following a presentation we both saw at IEEE where someone did that for ADAPT. So it would seem that if we were to monitor these quantities, basic VQE is not where it should happen :)